### PR TITLE
JSON Utility:Modify API to get offset for VPD path

### PR DESCRIPTION
**Pull request by PriyangaRamasamy**
This commit implements changes in the existing API getVPDOffset() in JSON utility file.

Current implementation of getVPDOffset is to get VPD offset value from JSON given only the primary EEPROM path(if present in JSON).

Modified implementation is to get VPD offset value from JSON given either primary EEPROM path or redundant EEPROM path(if present in JSON).

### CODE REVIEW

**Review by SunnySrivastava1984 — APPROVED** [review]

File: include/utility/json_utility.hpp [hunk 1]
```diff
@@ -47,29 +47,40 @@ inline std::unordered_map<std::string, functionPtr> funcionMap{
...
+
+    const nlohmann::json& l_fruList =
+        i_sysCfgJsonObj["frus"].get_ref<const nlohmann::json::object_t&>();
+
+    for (const auto& l_fru : l_fruList.items())
+    {
+        const auto l_fruPath = l_fru.key();
+
+        // check if given path is redundant FRU path
+        if (i_vpdFilePath == i_sysCfgJsonObj["frus"][l_fruPath].at(0).value(
```
Comment (branupama) [inline include/utility/json_utility.hpp:75]: why we are hardcoding & finding at '0' index only ?
In future, if the fru's order gets changed, then this check won't be able to find the 'redundantEeprom' tag.

**Review by branupama — CHANGES_REQUESTED** [review]

File: include/utility/json_utility.hpp [hunk 2]
```diff
@@ -47,29 +47,40 @@ inline std::unordered_map<std::string, functionPtr> funcionMap{
 /**
  * @brief API to read VPD offset from JSON file.
  *
- * @param[in] parsedJson - Parsed JSON file for the system.
- * @param[in] vpdFilePath - path to the VPD file.
+ * @param[in] i_sysCfgJsonObj - Parsed JSON file for the system.
```
Comment (jinuthomas) [inline include/utility/json_utility.hpp:50]: this is an object

File: include/utility/json_utility.hpp [hunk 3]
```diff
@@ -47,29 +47,40 @@ inline std::unordered_map<std::string, functionPtr> funcionMap{
...
+ * @param[in] i_vpdFilePath - path to the VPD file.
  * @return Offset of VPD in VPD file.
```
Comment (jinuthomas) [inline include/utility/json_utility.hpp:52]: what is a VPD file ?

**Review by jinuthomas — CHANGES_REQUESTED** [review]

File: include/utility/json_utility.hpp (hunk 1 above)
Comment (PriyangaRamasamy) [reply include/utility/json_utility.hpp:75]: fru's order never changes. and even if it changes, 0th index will have all the properties which are common for the rest of the subfrus under that eeprom.

File: include/utility/json_utility.hpp (hunk 3 above)
Comment (PriyangaRamasamy) [reply include/utility/json_utility.hpp]: modified the existing statement

File: include/utility/json_utility.hpp [hunk 4]
```diff
@@ -47,29 +47,45 @@ inline std::unordered_map<std::string, functionPtr> funcionMap{
...
 {
-    size_t vpdOffset = 0;
-    if (!vpdFilePath.empty())
+    if (i_vpdFilePath.empty())
     {
-        if (parsedJson["frus"].contains(vpdFilePath))
+        return 0;
+    }
+
+    if (i_sysCfgJsonObj.empty())
```
Comment (branupama) [inline include/utility/json_utility.hpp:62]: can we combine this check along with line no 57, as it only returning 0. ?

File: include/utility/json_utility.hpp [hunk 5]
```diff
@@ -47,29 +47,45 @@ inline std::unordered_map<std::string, functionPtr> funcionMap{
...
+    {
+        return 0;
+    }
+
+    if (i_sysCfgJsonObj["frus"].contains(i_vpdFilePath))
```
Comment (branupama) [inline include/utility/json_utility.hpp:67]: check for 'frus' exist first before directly accessing it.

**Review by branupama — CHANGES_REQUESTED** [review]

**Review by branupama — APPROVED** [review]

File: include/utility/json_utility.hpp (hunk 1 above)
Comment (jinuthomas) [reply include/utility/json_utility.hpp:76]: Can you not use contains for finding redundantEeprom?

**Review by jinuthomas — APPROVED** [review]

**Review by SunnySrivastava1984 — APPROVED** [review]

**Pull request merged**